graph/secret.go

```diff
@@ -27,6 +27,14 @@ type Secret struct {
 	Akv         string `yaml:"akv,omitempty"`
 	MsiClientID string `yaml:"clientID,omitempty"`
 
+	// InternalID is an ID generated by secrets not specified in Task yaml file.
+	// For instance, Secrets passed in for login to custom registry will generate internal IDs.
+	// Need a separate ID so that it does not conflict with user-defined IDs
+	InternalID string
```

Comment:
secret is an entity in task. it's very different from the registry credentials even though both can reference key vault. so I don't think it's a right way to extend this secret model to support registry credential.